Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -188,4 +209,37 @@ public struct Configuration {
         let allPaths = self.lintablePathsForPath(path)
         return allPaths.flatMap { File(path: $0) }
     }
+
+    public func configForFile(file: File) -> Configuration {
+        if  useNestedConfigs,
+            let containingDir = file.path?.stringByDeletingLastPathComponent {
+            return configForPath(containingDir)
+        }
+        return self
+    }
+}
+
+// Nested configs
+extension Configuration {
+    private func configForPath(path: String) -> Configuration {
+        let configSearchPath = path.stringByAppendingPathComponent(".swiftlint.yml")
+
+        // If a config exists and it isn't us, load and merge the configs
+        if configSearchPath != configPath &&
+            NSFileManager.defaultManager().fileExistsAtPath(configSearchPath) {
+            return merge(Configuration(path: configSearchPath, optional: false, silent: true))
+        }
+
+        // If we are not at the root path, continue down the tree
+        if path != rootPath {
+            return configForPath(path.stringByDeletingLastPathComponent)
+        }
+
+        // If nothing else, return ourselves
+        return self
+    }
+
+    private func merge(config: Configuration) -> Configuration {
```

Comment:
might be worth adding a note to eventually expand the logic to merge configurations rather than replace the current one.